src/odemis/gui/cont/acquisition.py

```diff
@@ -761,34 +733,39 @@ def _update_acquisition_time(self):
         self._panel.txt_cryosecom_est_time.SetLabel(txt)
 
     @call_in_wx_main
-    def _on_name_and_wavelength(self, VA_value):
+    def _on_stream_wavelength(self, _):
+        """
+        a VA callback that is called when the user changes
+         the emission or excitation wavelength
+        """
+        counts = self._panel.streams_chk_list.GetCount()
+        streams = [
+            self._panel.streams_chk_list.GetClientData(i) for i in range(counts)
+        ]
+        for i, s in zip(range(counts), acqmng.sortStreams(streams)):
+            # replace the unsorted streams with the sorted streams one by one
+            self._panel.streams_chk_list.Delete(i)
+            self._panel.streams_chk_list.Insert(s.name.value, i, s)
+            self._panel.streams_chk_list.Check(i)
```

Comment:
What does "Check()"? Does it toggle the stream? If so, that shouldn't always be the case. Only if it's in acquisitionStreams.

Reply:
Good catch